packages/@uppy/core/src/Uppy.ts

```diff
@@ -154,9 +154,12 @@ type MinimalRequiredOptions<M extends Meta, B extends Body> = Partial<
   }
 >
 
-export type NonNullableUppyOptions<M extends Meta, B extends Body> = Required<
-  UppyOptions<M, B>
->
+// everything is required, except for `locale`
+export type NonNullableUppyOptions<M extends Meta, B extends Body> = Omit<
+  Required<UppyOptions<M, B>>,
+  'locale'
+> &
+  Pick<UppyOptions<M, B>, 'locale'>
```

Comment:
We could probably be using the `DefinePluginOpts` utility type

Reply:
not sure if I did correctly, but I tried to change it to:

```ts
export type NonNullableUppyOptions<M extends Meta, B extends Body> = DefinePluginOpts<UppyOptions<M, B>, 'locale'>
```

then the effective type turns out to be:

```ts
type NonNullableUppyOptions<M extends Meta, B extends Body> = UppyOptions<M, B> & Required<Pick<UppyOptions<M, B>, "locale">>
```

however this caused a lot of type errors. looking at the current implementation, it yields this effective type instead:

```ts
type NonNullableUppyOptions<M extends Meta, B extends Body> = Omit<Required<UppyOptions<M, B>>, "locale"> & Pick<UppyOptions<M, B>, "locale">
```

diff:
```diff
- type NonNullableUppyOptions<M extends Meta, B extends Body> = Omit<Required<UppyOptions<M, B>>, "locale"> & Pick<UppyOptions<M, B>, "locale">
+ type NonNullableUppyOptions<M extends Meta, B extends Body> = UppyOptions<M, B> & Required<Pick<UppyOptions<M, B>, "locale">>
```